modules/tinymce/src/core/test/ts/browser/FormatterCheckTest.ts

```diff
@@ -260,4 +260,36 @@ describe('browser.tinymce.core.FormatterCheckTest', () => {
     editor.selection.setRng(rng);
     assert.isFalse(editor.formatter.match('link'), 'No match on named anchor');
   });
+
+  it('TINY-7227: match one class on an element with multiple classes', () => {
+    const editor = hook.editor();
+    editor.formatter.register('formatA', { selector: 'p', classes: [ '%value' ] });
+
+    editor.setContent('<p class="a b c">test</p>');
+    LegacyUnit.setSelection(editor, 'p', 0, 'p', 0);
+
+    assert.isTrue(editor.formatter.match('formatA', { value: 'b' }), 'Should match since the onmatch matches on "b" class.');
+  });
+
+  it('TINY-7227: match multiple values on an element with multiple classes', () => {
```

Comment:
Also add a test where it checks if match returns false since in theory that function could not always be returning true.